python/lsst/ap/pipe/matchSourceInjected.py

```diff
@@ -129,8 +134,79 @@ def run(self, injectedInitialPVICat, diffIm, associatedDiaSources):
         else:
             fakeCat = injectedInitialPVICat
 
+        self._estimateFakesSNR(fakeCat, diffIm)
+
         return self._processFakes(fakeCat, associatedDiaSources)
 
+    def _estimateFakesSNR(self, injectedCat, diffIm):
+        """Estimate the signal-to-noise ratio of the fakes in the given catalog.
+
+        Parameters
+        ----------
+        injectedCat : `astropy.table.Table`
+            Catalog of synthetic sources to estimate the S/N of. **This table
+            will be modified in place**.
+        diffIm : `lsst.afw.image.Exposure`
+            Difference image where the sources were detected.
+
+        Returns
+        -------
+        None
+        """
+        # Create a schema for the forced measurement task
+        schema = afwTable.SourceTable.makeMinimalSchema()
+        schema.addField("x", "D", "x position in image.", units="pixel")
+        schema.addField("y", "D", "y position in image.", units="pixel")
+        schema.addField("deblend_nChild", "I", "Need for minimal forced phot schema")
+
+        # Create the forced measurement task
+        forcedMeas = ForcedMeasurementTask(schema)
+        # Remove plugins that are not required for this task
+        forcedMeas.plugins.pop('base_SdssShape')
+        forcedMeas.plugins.pop('base_GaussianFlux')
+        forcedMeas.plugins.pop('base_TransformedCentroid')
+        forcedMeas.plugins.pop('base_TransformedShape')
+        # Specify the columns to copy from the input catalog to the output catalog
+        forcedMeas.copyColumns = {"coord_ra": "ra", "coord_dec": "dec"}
+
+        # Create an afw table from the input catalog
+        outputCatalog = afwTable.SourceCatalog(schema)
+        outputCatalog.reserve(len(injectedCat))
+        for row in injectedCat:
+            outputRecord = outputCatalog.addNew()
+            outputRecord.setId(row['injection_id'])
+            outputRecord.setCoord(SpherePoint(row["ra"], row["dec"], degrees))
+            outputRecord.set("x", row["x"])
+            outputRecord.set("y", row["y"])
+
+        # Generate the forced measurement catalog
+        forcedSources = forcedMeas.generateMeasCat(diffIm, outputCatalog, diffIm.getWcs())
+        # Attach the PSF shape footprints to the forced measurement catalog
+        forcedMeas.attachPsfShapeFootprints(forcedSources, diffIm)
+
+        # Copy the x and y positions from the forced measurement catalog back
+        # to the input catalog
+        for tgt, src in zip(outputCatalog, forcedSources):
+            src.set('base_SdssCentroid_x', tgt['x'])
```

Comment:
Doesn't this set the values in the forcedSources catalog to the values in outputCatalog?

Reply:
This is setting the values in the input measurement catalog to be the ones already set up in the output record as "x" and "y". 
Since these coordinates come from the truth catalog I think it is Ok.